cpp/src/parquet/stream_reader.cc

```diff
@@ -17,12 +17,31 @@
 
 #include "parquet/stream_reader.h"
 
+#include <set>
 #include <utility>
 
 namespace parquet {
 
 constexpr int64_t StreamReader::kBatchSizeOne;
 
+// The converted type expected by the stream reader does not always
+// exactly match with the schema in the Parquet file.  The following
+// is a list of converted types which are allowed instead of the
+// expected converted type.
+// Each pair given is:
+//   {<StreamReader expected type>, <Parquet file converted type>}
+// So for example {ConvertedType::INT_32, ConvertedType::NONE} means
+// that if the StreamReader was expecting the converted type INT_32,
+// then it will allow the Parquet file to use the converted type
+// NONE.
+//
+static const std::set<std::pair<ConvertedType::type, ConvertedType::type> >
+    converted_type_exceptions = {{ConvertedType::INT_32, ConvertedType::NONE},
```

Comment:
What is the rationale for these exceptions? For example, why `(INT64, NONE)` but not `(UINT8, NONE)`?